src/lib/mdns/Resolver.h

```diff
@@ -100,6 +103,66 @@ struct DiscoveredNodeData
     DiscoveredNodeData() { Reset(); }
     bool IsHost(const char * host) const { return strcmp(host, hostName) == 0; }
     bool IsValid() const { return !IsHost("") && ipAddress[0] != chip::Inet::IPAddress::Any; }
+    void Log() const
+    {
+        if (rotatingIdLen > 0)
+        {
+            char rotatingIdString[chip::Mdns::kMaxRotatingIdLen * 2 + 1] = "";
+            Encoding::BytesToUppercaseHexString(rotatingId, rotatingIdLen, rotatingIdString, sizeof(rotatingIdString));
+            ChipLogDetail(DiscoveryVerbose, "Rotating ID: %s", rotatingIdString);
+        }
+        if (strlen(deviceName) != 0)
+        {
+            ChipLogDetail(DiscoveryVerbose, "Device Name: %s", deviceName);
+        }
+        if (vendorId > 0)
+        {
+            ChipLogDetail(DiscoveryVerbose, "Vendor ID: %u", vendorId);
+        }
+        if (productId > 0)
+        {
+            ChipLogDetail(DiscoveryVerbose, "Product ID: %u", productId);
+        }
+        if (deviceType > 0)
+        {
+            ChipLogDetail(DiscoveryVerbose, "Device Type: %u", deviceType);
+        }
+        if (longDiscriminator > 0)
+        {
+            ChipLogDetail(DiscoveryVerbose, "Long Discriminator: %u", longDiscriminator);
+        }
+        if (!IsHost(""))
+        {
+            ChipLogDetail(DiscoveryVerbose, "Hostname: %s", hostName);
+        }
+        if (additionalPairing > 0)
+        {
+            ChipLogDetail(DiscoveryVerbose, "Additional Pairing: %u", additionalPairing);
+        }
+        if (strlen(pairingInstruction) != 0)
+        {
+            ChipLogDetail(DiscoveryVerbose, "Pairing Instruction: %s", pairingInstruction);
+        }
+        if (pairingHint > 0)
+        {
+            ChipLogDetail(DiscoveryVerbose, "Pairing Hint: 0x%x", pairingHint);
+        }
+        if (numIPs > 0)
+        {
+            for (int j = 0; j < numIPs; j++)
+            {
+                char ipAddressString[Inet::kMaxIPAddressStringLength];
+                ChipLogDetail(DiscoveryVerbose, "IP Address #%d: %s", j + 1,
+                              ipAddress[j].ToString(ipAddressString, sizeof(ipAddressString)));
+                (void) ipAddressString;
```

Comment:
Please separate the conversion from the logging, and remove `(void)ipAddressString`

Reply:
Separated